packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -410,106 +869,777 @@ class _CupertinoAlertActionSection extends StatelessWidget {
+
+  double _calculatePadding(BuildContext context) {
+    return 8.0 * MediaQuery.textScaleFactorOf(context);
+  }
+
+  // Dialog action content shrinks to fit, up to a certain point, and if it still
+  // cannot fit at the minimum size, the text content is ellipsized.
+  //
+  // This policy only applies when the device is not in accessibility mode.
+  Widget _buildContentWithRegularSizingPolicy({
+    @required BuildContext context,
+    @required TextStyle textStyle,
+    @required Widget content,
+  }) {
+    final bool isInAccessibilityMode = _CupertinoAccessibility.of(context)?.isInAccessibilityMode ?? false;
+    final double dialogWidth = isInAccessibilityMode
+        ? _kAccessibilityCupertinoDialogWidth
+        : _kCupertinoDialogWidth;
+    final double textScaleFactor = MediaQuery.textScaleFactorOf(context);
+    final double fontSizeRatio = (textScaleFactor * textStyle.fontSize) / _kMinButtonFontSize;
+    final double padding = _calculatePadding(context);
+
+    return new IntrinsicHeight(
+      child: new SizedBox(
+        width: double.infinity,
+        child: new FittedBox(
+          fit: BoxFit.scaleDown,
+          child: new ConstrainedBox(
+            constraints: new BoxConstraints(
+              maxWidth: fontSizeRatio * (dialogWidth - (2 * padding)),
+            ),
+            child: new DefaultTextStyle(
+              style: textStyle,
+              textAlign: TextAlign.center,
+              overflow: TextOverflow.ellipsis,
+              maxLines: 1,
+              child: content,
             ),
           ),
         ),
-      );
+      ),
+    );
+  }
+
+  // Dialog action content is permitted to be as large as it wants when in
+  // accessibility mode. If text is used as the content, the text wraps instead
+  // of ellipsizing.
+  Widget _buildContentWithAccessibilitySizingPolicy({
+    @required TextStyle textStyle,
+    @required Widget content,
+  }) {
+    return new DefaultTextStyle(
+      style: textStyle,
+      textAlign: TextAlign.center,
+      child: content,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    TextStyle style = _kCupertinoDialogActionStyle;
+    style = style.merge(textStyle);
+
+    if (isDestructiveAction) {
+      style = style.copyWith(color: CupertinoColors.destructiveRed);
+    }
+
+    if (!enabled) {
+      style = style.copyWith(color: style.color.withOpacity(0.5));
     }
+
+    // Apply a sizing policy to the action button's content based on whether or
+    // not the device is in accessibility mode.
+    // TODO(mattcarroll): The following logic is not entirely correct. It is also
+    // TODO(mattcarroll): the case that if content text does not contain a space,
+    // TODO(mattcarroll): it should also wrap instead of ellipsizing. We are
+    // TODO(mattcarroll): consciously not implementing that now due to complexity.
+    final Widget sizedContent = _CupertinoAccessibility.of(context)?.isInAccessibilityMode ?? false
```

Comment:
ah, interesting, i guess this is why you needed _CupertinoAccessibility...

Might be best to just have a function that takes a scale factor and tells you if you need accessibility mode, and to rely on MediaQuery instead of this private inherited widget, because:

What happens if someone uses this widget outside of where you'd expect? e.g. in a test case or demo or some such. Right now, it'll never correctly the determine accessibility mode.

Reply:
Switched the implementation from inherited widget to a top-level private function